storybook/src/stories/UseDeepCompareEffect/API/API.tsx

```diff
@@ -0,0 +1,35 @@
+import throttle from 'lodash.throttle'
+import React, { useRef, useState } from 'react'
+import { useDeepCompareEffect } from 'too-many-hooks'
+import { COUNTRY_CODES } from './constants'
+
+interface Props {
+  firstName: string
+  countryKey: keyof typeof COUNTRY_CODES
+}
+
+const API: React.FC<Props> = ({ firstName, countryKey }) => {
+  const [predictedAge, setPredictedAge] = useState<string>()
+
+  const getThrottledPrediction = useRef(
+    throttle(
+      (params: { name: string; country_id: typeof COUNTRY_CODES[keyof typeof COUNTRY_CODES] }) =>
+        fetch('https://api.agify.io?' + new URLSearchParams(params))
+          .then(res => res.json())
+          .then(data => setPredictedAge(data.age)),
+      2000,
+    ),
+  )
+
+  const fetchArgs = { name: firstName, country_id: COUNTRY_CODES[countryKey] }
+
+  useDeepCompareEffect(() => {
+    if (fetchArgs?.name && fetchArgs.country_id) {
+      getThrottledPrediction.current(fetchArgs)
+    }
+  }, [fetchArgs])
+
+  return <p className="prose">Predicted age: {predictedAge}</p>
```

Comment:
A few comments on the story:

- We should either color the text differently in dark mode or override and force light mode:

<img width="210" alt="image" src="https://user-images.githubusercontent.com/10625031/215298066-27e6bb6f-b599-4e42-a59f-e930663aa2f7.png">

- The text is a bit small on desktop. Maybe it could scale with the container?
<img width="776" alt="image" src="https://user-images.githubusercontent.com/10625031/215298141-e199d9aa-d3df-40b2-9016-e60fd6790269.png">

- It would be a helpful to have a description of what is happening, telling the user what is being shown and to use the controls to adjust the variables.

- Perhaps we could show some indicator that there's no predicted age instead of leaving it blank. Maybe an "invalid name" message?

Reply:
Yeah I will definitely make the text larger and add some description text. As far as the dark mode / light mode I was thinking #25 should fix the visibility issues so there is no point trying to build a fix or putting in overrides for now. If we think that ticket may we a way out though I can just override it to light mode.

Reply:
Waiting for #25 makes sense to me!